### ISSUE TITLE
Ngx Extended Pdf Viewer 404 (Not Found)

### ISSUE THREAD

**Issue by Valma-EG** [stephanrauh/ngx-extended-pdf-viewer#2358]
**I need some help**

![image](https://github.com/stephanrauh/ngx-extended-pdf-viewer/assets/164154346/dcd58622-d4b1-427b-a386-a330065dfd02)
My pdf viewer is no more working on local giving me this error. I've tried to solve it but I'm blocked and I don't know how to go on.

this is my html:

```
<div class="pdf-viewer">
    <ngx-extended-pdf-viewer 
        *ngIf="pdfSrc"
        [src]="pdfSrc" 
        [formData]="form?.value" 
        [textLayer]="true"
        (formDataChange)="formDataChanged($event)" 
        [disableForms]="disableAcroForms" 
        (pageRendered)="setupForm()"
        useBrowserLocale="true" 
        height="100vh">
    </ngx-extended-pdf-viewer>
</div>

```
where:
pdfSrc = 'assets/bu830-31.pdf';

**Version info**
Agular 17
 "ngx-extended-pdf-viewer": "^19.7.1"

**Comment by stephanrauh** [comment]
At the moment, I'm also confused. I decided to remove the minified ES5 versions of the files some time ago, but when I did, I also made sure these files aren't requested. Appearently, this didn't work.

While I'm still trying to figure it out, I suppose you can solve the issue by adding the attribute `[minifiedJSLibraries]="false"` to the `<ngx-extended-pdf-viewer>` tag.

**Comment by Valma-EG** [comment]
Ok, thank you, now I don't have the first error on the op-chaining-support.js, but still on the other:

![image](https://github.com/stephanrauh/ngx-extended-pdf-viewer/assets/164154346/b128dd05-96d3-4c9e-b688-ba5b1901d87e)

So the pdf still not be charged.

**Comment by Valma-EG** [comment]
I just add this line in the constructor and now seems it works:

![image](https://github.com/stephanrauh/ngx-extended-pdf-viewer/assets/164154346/ff8a60f6-a1e2-47cf-bab8-26df5b20015d)

**Comment by stephanrauh** [comment]
@Valma-EG Did you check the URL of the missing files? Maybe the PDF loader tries to load the files from the wrong directory.

**Comment by ArsenijeD** [comment]
@Valma-EG @stephanrauh 

Can you summarize at the end what is needed to be done, in order to work around this issue?
Because, currently I cannot figure out from the previous discussion.

**Comment by Valma-EG** [comment]
I resolved the issue in this way:
https://github.com/stephanrauh/ngx-extended-pdf-viewer/issues/2358#issuecomment-2149159703

**Comment by ArsenijeD** [comment]
> I resolved the issue in this way:
> https://github.com/stephanrauh/ngx-extended-pdf-viewer/issues/2358#issuecomment-2149159703

Okay, can you tell me about the  property `pdfDefaultOptions`, where is it used?

**Comment by stephanrauh** [comment]
`pdfDefaultOptions` is a global variable allowing you to configure the behavior of the core viewer. It's documented here: https://pdfviewer.net/extended-pdf-viewer/options

**Comment by stephanrauh** [comment]
I've closing the ticket because it seems to be solved.

**Comment by dsdatta** [comment]
@stephanrauh @Valma-EG @B34v0n @ArsenijeD @topce 
For Info, incase someone is the experiencing the same issue.
I got the same error. I was using angular 17 with ngx-pdf-viwer 22.3.9 backend .net core . In the production environment i was having 404 not found for .mjs and.ftl file. whereas the local environment was working fine  local IIS was working fine.
This solved the issue.

check your IIS logs and see if it has a specific error - 404.7, If so then you might need to add the below in you web.config file
<security>
   <requestFiltering removeServerHeader="true">
     <fileExtensions>    
       <add fileExtension=".mjs" allowed="true" />
       <add fileExtension=".ftl" allowed="true" />
     </fileExtensions>
   </requestFiltering>
 </security>


https://stackoverflow.com/a/79426553/29579582

**Comment by SakamotoLfe** [comment]
Totally late for the party and probably giving the wrong answer BUT, as someone who's desperate I've found this solution:

Adding this snippet of code to `angular.json` inside the `assets` solved it for me:

`{
    "glob": "**/*",
    "input": "node_modules/ngx-extended-pdf-viewer/assets/",
    "output": "/assets/ngx-extended-pdf-viewer/"
}`

The `assets` ended up like this:

`"assets": [
              "src/favicon.ico",
              "src/assets",
              {
                "glob": "**/*",
                "input": "node_modules/ngx-extended-pdf-viewer/assets/",
                "output": "/assets/ngx-extended-pdf-viewer/"
              }
            ],`

I'm using Angular 16 and "ngx-extended-pdf-viewer": "^19.7.1", hope this helps someone

**Comment by stephanrauh** [comment]
Also see https://pdfviewer.net/extended-pdf-viewer/getting-started and click the second tab to see the `angular.json`.
